pkg/gatewayserver/io/mqtt/mqtt.go

```diff
@@ -95,9 +95,11 @@ type connection struct {
 func (*connection) Protocol() string            { return "mqtt" }
 func (*connection) SupportsDownlinkClaim() bool { return false }
 
-func (c *connection) setup(ctx context.Context) error {
+func (c *connection) setup(ctx context.Context) (err error) {
 	ctx = auth.NewContextWithInterface(ctx, c)
-	defer recoverMQTTFrontend(ctx)
+	defer func() {
+		err = recoverMQTTFrontend(ctx)
```

Comment:
Aren't we now setting `err` to `nil` when there's no panic?

Reply:
Ah s*** yeah good catch.